sdk/servicebus/azure-servicebus/azure/servicebus/aio/_transport/_uamqp_transport_async.py

```diff
@@ -213,8 +207,8 @@ async def iter_next_async(
                 receiver._receive_context.set()
                 await receiver._open()
                 if not receiver._message_iter:
-                    receiver._message_iter = receiver._handler.receive_messages_iter_async()
-                uamqp_message = await receiver._message_iter.__anext__()
+                    receiver._message_iter = cast("ReceiveClientAsync", receiver._handler).receive_messages_iter_async()
+                uamqp_message = await cast(AsyncIterator, receiver._message_iter).__anext__()
```

Comment:
Is there a way  to not cast this, otherwise looks good!

Reply:
I can't think of one :( I could cast a line earlier, but b/c `receive._message_iter` can be None, the original type is Optional[AsyncIterator[Message]]]. So at some point, this will need to be cast to AsyncIterator so that we don't get the `None has no attribute __anext__` error.